components/script/dom/gpudevice.rs

```diff
@@ -636,6 +650,86 @@ impl GPUDeviceMethods for GPUDevice {
         GPUCommandEncoder::new(&self.global(), self.channel.clone(), encoder, true)
     }
 
+    /// https://gpuweb.github.io/gpuweb/#dom-gpudevice-createtexture
+    fn CreateTexture(&self, descriptor: &GPUTextureDescriptor) -> DomRoot<GPUTexture> {
+        let mut valid = true;
+        let size = match descriptor.size {
```

Comment:
you'll likely need that conversion in other places too, like in copy_buffer_to_texture and such